### PR TITLE
DDP-5509 support filtering questions to copy from previous instance

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Implements DDP-5509. This PR adds ability to specific a limited set of questions to copy answers from previous instance. One use-case is in #582.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/CopyConfigurationDao.sql.stg [hunk 1]
```diff
@@ -30,12 +30,20 @@ select cfg.copy_configuration_id,
...
+findPreviousInstanceFiltersByConfigId() ::= <<
+select filter.*,
+       loc.question_stable_code_id,
+       (select stable_id from question_stable_code
+         where question_stable_code_id = loc.question_stable_code_id
+       ) as question_stable_id
+  from copy_previous_instance_filter as filter
+  join copy_answer_location as loc on loc.copy_location_id = filter.answer_location_id
+ where filter.copy_configuration_id = :configId
+ order by filter.execution_order asc
```
Comment (ssettipalli) [inline pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/CopyConfigurationDao.sql.stg:48]: ok. this is how we are using the execution_order

File: pepper-apis/src/main/java/org/broadinstitute/ddp/copy/CopyExecutor.java [hunk 2]
```diff
@@ -125,6 +125,17 @@ private void copyAnswersFromPreviousInstance(Handle handle, CopyConfiguration co
         for (var childStableIds : childQuestionStableIds.values()) {
             questionStableIds.addAll(childStableIds);
         }
+
+        Set<String> specifiedStableIds = config.getPreviousInstanceFilters().stream()
```
Comment (ssettipalli) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/copy/CopyExecutor.java:154]: How is the execution_order retained from here on !

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/copy/CopyExecutor.java:154]: Thanks for the catch! It wasn't respecting execution order of these filters but I have refactored to do so!

Comment (ssettipalli) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/copy/CopyExecutor.java:154]: :+1:

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**